### ISSUE TITLE
Add support for Ignition v3

### ISSUE THREAD

**Issue by johananl** [flatcar/Flatcar#942]
Allow CAPI users to leverage Ignition v3 when bootstrapping Flatcar nodes.

**Comment by vincepri** [comment]
Have we considered adopting butane config types in the bootstrap provider / machine resource in Cluster API?

**Comment by johananl** [comment]
I'd have to look at the specifics to have an informed opinion here. Care to elaborate @vincepri? 🙂

**Comment by johananl** [comment]
(I assume this is following [this](https://kubernetes.slack.com/archives/C03GQ8B5XNJ/p1684243738471749) Slack thread but I'm not clear on the details.)

**Comment by vincepri** [comment]
Yep! Maybe let's sync up offline at some point and come up with a plan / CAEP 🙃

**Comment by johananl** [comment]
Relevant discussion: https://kubernetes.slack.com/archives/C8TSNPY4T/p1688643597808979

**Comment by johananl** [comment]
Upstream PR: https://github.com/kubernetes-sigs/cluster-api/pull/9158
